loris/webapp.py

```diff
@@ -517,7 +517,10 @@ def _get_info(self,ident,request,base_uri):
         else:
             in_cache = False
 
-        if in_cache:
+        #Checking for src_format in ImageInfo signals that it's not old cache data:
+        #   src_format should be in the cache now.
+        #   TODO: remove in Loris 4.0.
+        if in_cache and self.info_cache[request][0].src_format:
```

Comment:
A few questions:

* The old cache data was `info.to_json(cache=True)`, which would presumably include `src_format`? So is this addressing an earlier time where it wasn't included?

* As Loris is 2.1.0 now, next major release is 3.0 – so why wait until 4.0 to make breaking cache changes?

* To be clear: what we'd be removing in 4.0 is the second half of this check? Maybe that should be explicit in the comment?

Meta-comment: I didn't immediately recognise the format of `self.info_cache[request]` – I wonder whether we could store namedtuples in InfoCache for greater clarity here?

Reply:
@alexwlchan thanks for your review. I pushed another commit. As far as whether to remove the cache check in 3.0 or 4.0, @giorgiosironi had a comment about it in https://github.com/loris-imageserver/loris/issues/356#issuecomment-330505838. @giorgiosironi what are your thoughts about removing the cache check in a 3.0 or a 4.0 release?

I'm OK either way. Regardless, we'll have to let users know that they'll have to make sure their cache doesn't have any of the old content once we remove the check.

Reply:
Ah, I’d missed the idea of making 3.0 a Python 3-compatible release, and deferring the cache changes until later. I don’t mind too much when we change it, I just wanted to check if I’d forgotten another conversation, and I had.

Important thing is that we have a note to come back and change this later, which we do! :-)